src/routes/expenses.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+const express = require('express');
+const expenseController = require('../controllers/expenses');
+
+const router = express.Router();
+
+const hasParams = (req, _res, next) => {
+  if (Object.keys(req.query).length > 0) {
+    next();
+  } else {
+    next('route');
+  }
+};
+
+router.get('/', hasParams, expenseController.getByParameters);
```

Comment:
Good work figuring out a custom solution using next('route') :1st_place_medal:. Unfortunately, introducing custom logic to your routing is error-prone and can get real complicated in big projects.

Move this logic to the controller (use only one controller method). This is okay, as `.getAll` is only 2 lines long.